### PR TITLE
add checknameavailability

### PR DESCRIPTION
**Pull request by tarostok**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
Thanks @tarostok,
The spec is currently invalid - could you take a look?
```
ERROR: Schema violation: Additional properties not allowed: /subscriptions/{subscriptionId}/providers/Microsoft.PowerBIDedicated/locations/{location}/checkNameAvailability
    - file:///tmp/tmp2octhxb6/rest/specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/powerbidedicated.json:1:0 ($)
ERROR: Schema violation: Data does not match any schemas from 'oneOf'
    - file:///tmp/tmp2octhxb6/rest/specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/powerbidedicated.json:820:4 ($.parameters.CheckCapacityNameAvailabilityResult)
ERROR: Schema violation: Data does not match any schemas from 'oneOf'
    - file:///tmp/tmp2octhxb6/rest/specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/powerbidedicated.json:802:4 ($.parameters.CheckCapacityNameAvailabilityParameters)
```

**Comment by sergey-shandar** [comment]
@annatisch you can assign the PR to me.

**Comment by tarostok** [comment]
Fixed spec.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/examples/checkNameAvailability.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "parameters": {
+    "subscriptionId": "613192d7-503f-477a-9cfe-4efc3ee2bd60",
+    "location": "West US",
+    "serverParameters": {
+      "name": "azsdktest",
+      "type": "Microsoft.PowerBIDedicated/capacities"
+    },
+    "api-version": "2017-10-01"
+  },
```
[the remark this answers is not in the document]
Comment (tarostok) [reply specification/powerbidedicated/resource-manager/Microsoft.PowerBIdedicated/stable/2017-10-01/examples/checkNameAvailability.json]: right, thanks, fixed the typo

**Review by sergey-shandar — APPROVED** [review]
@tarostok could you add codeowners for PowerBI dedicated to this file https://github.com/Azure/azure-rest-api-specs/blob/master/CODEOWNERS ? We need codeowners to request review if someone else create a PR against your RP.

**Comment by tarostok** [comment]
@sergey-shandar updated code owners

**Pull request merged**